packages/apis/ipfs/src/query/index.ts

```diff
@@ -0,0 +1,46 @@
+import { Http_Query, Http_ResponseType, Http_Body } from "./w3/imported";
+import { 
+  Input_catToString,
+  Input_catFile
+} from "./w3";
+import {decode} from "as-base64"
+
+const EmptyBody: Http_Body = {rawBody:null, stringBody:null, formDataBody: {data: []}}
+
+export function catToString(input: Input_catToString): String {
+  const url = input.ipfsUrl + "/api/v0/cat";
+  const catResponse = Http_Query.get({
+    url: url,
+    request: {
+      headers: [],
+      urlParams: [{key: "arg", value: input.cid}],
+      responseType: Http_ResponseType.TEXT,
+      body: EmptyBody,
+    }
+  });
+
+  if(catResponse == null || catResponse.status != 200) {
+    throw new Error(`Failed to cat file: ${catResponse?.status} ${catResponse?.statusText}`);
+  }
+
+  return catResponse.body;
+}
+
+export function catFile(input: Input_catFile): ArrayBuffer {
+  const url = input.ipfsUrl + "/api/v0/cat";
+  const catResponse = Http_Query.get({
+    url: url,
+    request: {
+      headers: [],
+      urlParams: [{key: "arg", value: input.cid}],
+      responseType: Http_ResponseType.BINARY,
+      body: EmptyBody,
+    }
+  });
+
+  if(catResponse == null || catResponse.status != 200) {
+    throw new Error(`Failed to cat file: ${catResponse?.status} ${catResponse?.statusText}`);
```

Comment:
I think a good standard error message pattern to use might be:
`IPFS catFile Failed.\nStatus: ${status}\nMessage: ${statusText}`

Reply:
I agree on the format, so I created a custom `IpfsError` for this, as I think that is the best way to handle this uniformly:

```ts
export class IpfsError extends Error { 
    constructor(method: string, status: number, statusText: string) {
      super(`IPFS ${method} Failed.\nStatus: ${status}\nMessage:${statusText}`);
      this.name = "IPFSError"
    }
}
```